app/src/main/java/com/example/bigowlapp/utils/SupervisorSchedulesAlarmManager.java

```diff
@@ -0,0 +1,85 @@
+package com.example.bigowlapp.utils;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+
+import com.example.bigowlapp.model.Schedule;
+import com.example.bigowlapp.repository.ScheduleRepository;
+import com.example.bigowlapp.service.SupervisorSchedulesAlarmReceiver;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * The purpose of this class is to set/define alarms for supervisor schedules that the app will set.
+ * After the time activation of an alarm, code will be executed from
+ * {@link SupervisorSchedulesAlarmReceiver}
+ */
+public class SupervisorSchedulesAlarmManager {
+
+
+    private final Context context;
+    private final ScheduleRepository scheduleRepository;
+    private final FirebaseFirestore db = FirebaseFirestore.getInstance();
+
+    public SupervisorSchedulesAlarmManager(Context context) {
+
+        this.context = context;
+        scheduleRepository = new ScheduleRepository();
+    }
+
+    /**
+     * Sets the alarm(s) for the SupervisorSchedulesAlarmReceiver given the coming schedules that the user is supervising
+     *
+     * @param userID The Id of the user who is a supervisor of some schedules
+     */
+    public void setAlarms(String userID) {
+        int requestCode = 0;
+        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        getSchedulesForSupervisor(userID)
+                .addOnSuccessListener(scheduleList -> {
+                    for (Schedule schedule : scheduleList) {
+                        Intent intent = new Intent(context, SupervisorSchedulesAlarmReceiver.class);
+                        intent.putExtra("scheduleUid", schedule.getUid());
+                        PendingIntent pendingIntent = PendingIntent
+                                .getBroadcast(context, requestCode, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+                        alarmManager.set(AlarmManager.RTC_WAKEUP,
+                                addMinutesToDate(schedule.getStartTime().toDate(), 4).getTime(), pendingIntent);
```

Comment:
I feel like having another AlarmManger  for this is unnecessary and the code could just be extracted to the `MemberScheduleAlarmManager`.

The reason I say this is because both managers run at the exact same place (HomepageActivity), at the exact same time, use the exact same inputs, use the exact same data (users schedules), and the only difference is that a different alarm is being set.

I don't think a separate AlarmManger is necessary to do the exact same thing and just make another alarm because you could just make the 2 alarms within the same manager easily, and reduce bloat as well as reduce database calls. It's also much simpler and will be easier to maintain.

I also think it would be more logical, the idea that one alarm manger makes 2 alarms for schedules, one for the detection, and another for the four minute later alarm you've implemented.

The second broadcast reciever on the other hand I think is fine, as the things these two alarms two is quite different and separating them out as different receivers I do think is valid.

If I missed something, as to why a second alarm manger is necessary, you can explain, as it might be possible there is something I didn't notice as to why a different manager is needed for this second alarm?

Thoughts? @ShawnGuerra-Bautista @Landry333

Reply:
Yeah, I agree with this point. We don't really need another manager and we can just use the same alarm manager to set 2 different alarms.

Reply:
The setAlarms(String userID) method in the SupervisorSchedulesAlarmManager does something different. It calls for getSchedulesForSupervisor(String userID) which is a different kind of database call compared to the getSchedulesForUser(String userID). Then the rest of the setAlarms(String userID) method in the SupervisorSchedulesAlarmManager is also a bit different

Reply:
@Landry333 Actually, yeah, I see what you mean. The db calls are different, Although I would still suggest you could do both db calls and alarms in one manager.

Reply:
ok